### PR TITLE
Rename external plugin to agent

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
as title

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
     },
     install_requires=[
         "googleapis-common-protos>=1.57",
-        "flyteidl>=1.5.6",
+        "flyteidl==1.5.9",
```
Comment (wild-endeavor) [inline setup.py:32]: wait til tests pass
```suggestion
        "flyteidl>=1.5.9,<1.6.0",
```

**Review by wild-endeavor — DISMISSED** [review]

File: setup.py (hunk 1 above)
Comment (pingsutw) [reply setup.py:32]: it doesn't work, but `pip install "flyteidl>=1.5.8,<1.6.0"` works.

Comment (pingsutw) [reply setup.py:32]: maybe merge this [pr](https://github.com/flyteorg/flyteidl/pull/411), then update it to "flyteidl>=1.5.9,<1.6.1",

File: setup.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
     },
     install_requires=[
         "googleapis-common-protos>=1.57",
-        "flyteidl>=1.5.6",
+        "flyteidl>1.5.6",
```
Comment (eapolinario) [inline setup.py:32]: this should be `flyteidl>=1.5.9`, no?

File: .github/workflows/pythonpublish.yml [hunk 3]
```diff
@@ -161,12 +161,12 @@ jobs:
           registry: ghcr.io
           username: "${{ secrets.FLYTE_BOT_USERNAME }}"
           password: "${{ secrets.FLYTE_BOT_PAT }}"
-      - name: Prepare External Plugin Service Image Names
-        id: external-plugin-service-names
+      - name: Prepare Fylte Agent Image Names
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply .github/workflows/pythonpublish.yml]: good catch, thanks

File: setup.py (hunk 2 above)
Comment (pingsutw) [reply setup.py]: bumped version to 1.5.10. we yanked 1.5.9

**Review by eapolinario — APPROVED** [review]

**Pull request merged**